### PR TITLE
Raise error for missing focal length

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Comment by LukasNickel** [comment]
Is 0 a default we expect somewhere?
I would have assumed to check `if not focal_length: ...`

**Comment by HealthyPear** [comment]
> Is 0 a default we expect somewhere? I would have assumed to check `if not focal_length: ...`

yes here
https://github.com/cta-observatory/ctapipe/blob/3e35093f2d2ade576c79b4b326bbab7b5bde298f/ctapipe/coordinates/camera_frame.py#L78

**Comment by maxnoe** [comment]
Astropy would complain about `None` and checking `np.isnan()` is much slower than checking `== 0`, so I left it in.

**Review by HealthyPear — CHANGES_REQUESTED** [review]
also the reverse transformation requires to know the focal length

https://github.com/cta-observatory/ctapipe/blob/3e35093f2d2ade576c79b4b326bbab7b5bde298f/ctapipe/coordinates/camera_frame.py#L176

File: ctapipe/coordinates/camera_frame.py [hunk 1]
```diff
@@ -174,6 +178,8 @@ def telescope_to_camera(telescope_coord, camera_frame):
         y_rotated = x_pos * sinrot + y_pos * cosrot
 
     focal_length = camera_frame.focal_length
+    if focal_length.value == 0:
```
Comment (maxnoe) [inline ctapipe/coordinates/camera_frame.py:181]: @HealthyPear here it is

Comment (HealthyPear) [reply ctapipe/coordinates/camera_frame.py:181]: oh right sorry I overlooked

**Review by HealthyPear — APPROVED** [review]

**Review by LukasNickel — APPROVED** [review]

**Pull request merged**